src/interaction/allowed_branches.go

```diff
@@ -0,0 +1,200 @@
+package interaction
+
+import (
+	"fmt"
+	"github.com/spf13/cast"
+	"keboola-as-code/src/model"
+	"keboola-as-code/src/utils"
+	"math"
+	"strings"
+)
+
+const (
+	ModeMainBranch     = "only main branch"
+	ModeAllBranches    = "all branches"
+	ModeSelectSpecific = "select branches"
+	ModeTypeList       = "type IDs or names"
+)
+
+type branchesPrompt struct {
+	prompt      *Prompt
+	allBranches []*model.Branch
+	isFlagSet   bool
+	flagValue   string
+	defaultMode interface{}
+}
+
+func (p *Prompt) GetAllowedBranches(allBranches []*model.Branch, isFlagSet bool, flagValue string) model.AllowedBranches {
+	// Convert CLI values to internal
+	if flagValue == "*" {
+		flagValue = model.AllBranchesDef
+	} else if flagValue == "main" {
+		flagValue = model.MainBranchDef
+	}
+
+	return (&branchesPrompt{
+		prompt:      p,
+		allBranches: allBranches,
+		isFlagSet:   isFlagSet,
+		flagValue:   flagValue,
+	}).ask()
+}
+
+func (p *branchesPrompt) ask() model.AllowedBranches {
+	if p.prompt.Interactive && !p.isFlagSet {
+		for {
+			results := p.doAsk()
+
+			// Check that the definition meets at least one branch
+			matched := 0
+			for _, branch := range p.allBranches {
+				for _, definition := range results {
+					if definition.IsBranchAllowed(branch) {
+						matched++
+					}
+				}
+			}
+
+			// If a branch is matched -> ok
+			if matched > 0 {
+				if matched == 1 {
+					p.prompt.Printf("\nOne branch match \"allowed branches\". Ok.\n")
+				} else {
+					p.prompt.Printf("\n%d branches match \"allowed branches\". Ok.\n", matched)
+				}
+				return results
+			}
+
+			p.prompt.Printf("\nNo existing branch matches your definitions. Please try again.\n")
+		}
+	}
+
+	// Parse flag value
+	return p.parseString(p.flagValue, ",")
+}
+
+func (p *branchesPrompt) doAsk() model.AllowedBranches {
+	// Ask user for mode
+	mode, ok := p.prompt.Select(&Select{
+		Label:       "Allowed branches",
+		Description: "Please select which branches you want to use with this CLI tool.",
+		Options: []string{
+			ModeMainBranch,
+			ModeAllBranches,
+			ModeSelectSpecific,
+			ModeTypeList,
+		},
+		Default: p.defaultMode,
+	})
+
+	// If it is necessary to repeat the selection (an error occurs), the option will be used as default
+	p.defaultMode = mode
+
+	// Load definitions according to the specified mode
+	if ok {
+		switch mode {
+		case ModeMainBranch:
+			return model.AllowedBranches{model.MainBranchDef}
+		case ModeAllBranches:
+			return model.AllowedBranches{model.AllBranchesDef}
```

Comment:
![image](https://user-images.githubusercontent.com/19371734/127662649-30de8e0c-3dea-40e6-98bf-f1e7733cde85.png)

Reply:
mas pravdu, vyhodim to v dalsom PR